src/jumpstart/JumpstartEnterAmount.test.tsx

```diff
@@ -249,5 +253,126 @@ describe('JumpstartEnterAmount', () => {
+    expect(getByText('jumpstartIntro.addFundsCelo.cta')).toBeTruthy()
+    expect(getByTestId('JumpstartAddAssets/noFundsButton')).toBeTruthy()
+  })
+
+  it('should show intro screen when user visits jumpstart for the first time', async () => {
+    const { getByText, getByTestId } = render(
+      <Provider
+        store={createMockStore({
+          tokens: { tokenBalances },
+          jumpstart: { introHasBeenSeen: false },
+        })}
+      >
+        <JumpstartEnterAmount />
+      </Provider>
+    )
+
+    expect(getByText('jumpstartIntro.title')).toBeTruthy()
+    expect(getByText('jumpstartIntro.description')).toBeTruthy()
+    expect(getByText('jumpstartIntro.haveFundsButton')).toBeTruthy()
+    expect(getByTestId('JumpstartEnterAmount/haveFundsButton')).toBeTruthy()
+  })
+
+  it('should track in analytics when user sees intro for the first time', async () => {
+    const updatedStore = createMockStore({
+      tokens: { tokenBalances },
+      jumpstart: { introHasBeenSeen: false },
+    })
+
+    const { getByText, getByTestId, rerender } = render(
+      <Provider store={updatedStore}>
+        <JumpstartEnterAmount />
+      </Provider>
+    )
+
+    expect(getByText('jumpstartIntro.title')).toBeTruthy()
+    expect(getByText('jumpstartIntro.description')).toBeTruthy()
+    expect(getByText('jumpstartIntro.haveFundsButton')).toBeTruthy()
+    expect(getByTestId('JumpstartEnterAmount/haveFundsButton')).toBeTruthy()
+    expect(AppAnalytics.track).toHaveBeenCalledWith(JumpstartEvents.jumpstart_intro_seen)
+
+    rerender(
+      <Provider store={updatedStore}>
+        <JumpstartEnterAmount />
+      </Provider>
+    )
+
+    expect(AppAnalytics.track).toBeCalledTimes(1)
+  })
+
+  it('should show intro screen every time until user clicks cta', async () => {
+    const updatedStore = createMockStore({
+      tokens: { tokenBalances },
+      jumpstart: { introHasBeenSeen: false },
+    })
+
+    const { getByText, getByTestId, rerender } = render(
+      <Provider store={updatedStore}>
+        <JumpstartEnterAmount />
+      </Provider>
+    )
+
+    expect(getByText('jumpstartIntro.title')).toBeTruthy()
+    expect(getByText('jumpstartIntro.description')).toBeTruthy()
+    expect(getByText('jumpstartIntro.haveFundsButton')).toBeTruthy()
+    expect(getByTestId('JumpstartEnterAmount/haveFundsButton')).toBeTruthy()
+
+    rerender(
+      <Provider store={updatedStore}>
+        <JumpstartEnterAmount />
+      </Provider>
+    )
+
+    // do not expect for jumpstartIntroSeen() to run
+    expect(updatedStore.getActions()).toEqual([depositTransactionFlowStarted()])
+
+    expect(getByText('jumpstartIntro.title')).toBeTruthy()
+    expect(getByText('jumpstartIntro.description')).toBeTruthy()
+    expect(getByText('jumpstartIntro.haveFundsButton')).toBeTruthy()
+    expect(getByTestId('JumpstartEnterAmount/haveFundsButton')).toBeTruthy()
+  })
+
+  it('should proceed to enter amount screen once the intro is seen', async () => {
```

Comment:
nice tests! i think one thing to call out is generally in the component tests, we've not tested for state changes (with `.getState()`) and instead relied on asserting that the correct actions have been received by the store. the state changes are driven by the reducers, which are not considered part of the component :) it's of course fine to include testing the reducer logic in components, but it's not really necessary